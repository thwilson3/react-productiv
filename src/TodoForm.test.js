import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoForm from "./TodoForm";
import { initialTodos } from "./App";


describe("TodoForm", function () {
    it("renders without crashing", function () {
        // this is a low-value test, but better than nothing
        render(<TodoForm />);
    });

    it("is blank if no form data", function () {
        const { container } = render(<TodoForm />);

        const title = container.querySelector("#newTodo-title");
        const description = container.querySelector("#newTodo-description");
        const priority = container.querySelector("#newTodo-priority");

        expect(title.value).toEqual("");
        expect(description.value).toEqual("");
        expect(priority.value).toEqual("1");

    });

    it("is populated with given form data", function () {

        const initialFormData = initialTodos[0]

        const { container } = render(
            <TodoForm initialFormData={initialFormData} />);

        const title = container.querySelector("#newTodo-title");
        const description = container.querySelector("#newTodo-description");
        const priority = container.querySelector("#newTodo-priority");

        expect(title.value).toEqual(initialFormData.title);
        expect(description.value).toEqual(initialFormData.description);
        expect(priority.value).toEqual(initialFormData.priority + "");

    });

    it("clears on submission", function () {

        const initialFormData = initialTodos[0]

        const { container } = render(
            <TodoForm
                initialFormData={initialFormData}
                handleSave={function () { }} />);

        const title = container.querySelector("#newTodo-title");
        const description = container.querySelector("#newTodo-description");
        const priority = container.querySelector("#newTodo-priority");

        const submit = container.querySelector("button");

        expect(title.value).toEqual(initialFormData.title);
        expect(description.value).toEqual(initialFormData.description);
        expect(priority.value).toEqual(initialFormData.priority + "");

        fireEvent.click(submit)

        expect(title.value).toEqual("");
        expect(description.value).toEqual("");
        expect(priority.value).toEqual("1");

    });

    it("matches snapshot", function () {
        const { container } = render(<TodoForm />);
        expect(container).toMatchSnapshot();
    });
});