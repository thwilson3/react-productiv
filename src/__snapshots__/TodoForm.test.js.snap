// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm matches snapshot 1`] = `
<div>
  <form
    class="NewTodoForm"
    id="todo-form-new"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      />
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;
